### ISSUE TITLE
Debug mode doesn't show gzipped bodies correctly

### ISSUE THREAD

**Issue by lucasrangeljs** [mgp25/Instagram-API#2649]
### Before submitting an issue make sure you have:
- [x] [Searched](https://github.com/mgp25/Instagram-API/search?type=Issues) the bugtracker for similar issues including **closed** ones
- [x] [Read the FAQ](https://github.com/mgp25/Instagram-API/wiki/FAQ)
- [x] [Read the wiki](https://github.com/mgp25/Instagram-API/wiki)
- [x] [Reviewed the examples](https://github.com/mgp25/Instagram-API/tree/master/examples)
- [x] [Installed the api using ``composer``](https://github.com/mgp25/Instagram-API#installation)
- [ ] [Using latest API release](https://github.com/mgp25/Instagram-API/releases)

### Purpose of your issue?
- [x] Bug report (encountered problems/errors)
- [ ] Feature request (request for new functionality)
- [ ] Question
- [ ] Other

---

I found a possible bug when I had debugged the data sent from API to Instagram. When `isBodyCompressed` is true, the post data isn't sent to Instagram.  

The error is at the end of the function `_getRequestBody` in the file `Request.php`:
```php
        if ($this->_isBodyCompressed) {
            return stream_for(zlib_encode((string) $this->_body, ZLIB_ENCODING_GZIP));
        }
        return $result;
```

If I'm not wrong, we need to replace `$this->_body` by `$result`:

```php
        if ($this->_isBodyCompressed) {
            return stream_for(zlib_encode((string) $result, ZLIB_ENCODING_GZIP));
        }
        return $result;
```

**Comment by JRoy** [comment]
The post data **is** sent to Instagram, debug mode is just unable to view the gzipped contents as of now.

I will add this functionality in a bit.

Cheers.

**Comment by lucasrangeljs** [comment]
Thanks for your fast reply.
But at this part of the code, we know that `$this->_body` is null, right? (because if it wasn't null, the function it would stopping at line 647 or 650 - first IF).

The data is in `$result` and not in `$this->_body`. Or am I wrong?

**Comment by JRoy** [comment]
No, if you look at the code `_body` isn't null:
https://github.com/mgp25/Instagram-API/blob/master/src/Request.php#L645-L651

Can you better explain what you're saying?

**Comment by lucasrangeljs** [comment]
Sorry for my bad explanation (English isn't my first language).

In the section (from line 652 to 671), we know that _body is null: 
https://github.com/mgp25/Instagram-API/blob/master/src/Request.php#L652-L671

there isn't a bug at the line 647 (here, `_body` isn't null).
but there is a bug at the line 668  (here, `_body` is null).

I've seen that you add two IF's in this function to implement the feature (compressed data).

In the first, you replaced: 
```php
return $this->_body;
```
by:
```php
if ($this->_isBodyCompressed) {
    return stream_for(zlib_encode((string) $this->_body, ZLIB_ENCODING_GZIP));
}
return $this->_body;
```

In the second, you replaced:
```php
return $result;
```
by:
```php
if ($this->_isBodyCompressed) {
    return stream_for(zlib_encode((string) $this->_body, ZLIB_ENCODING_GZIP));
}
return $result;
```

I've realized that the second must be:

```php
if ($this->_isBodyCompressed) {
    return stream_for(zlib_encode((string) $result, ZLIB_ENCODING_GZIP));
}
return $result;
```

**Comment by JRoy** [comment]
Thanks for the catch!

Fixed in 40fbbac0483c905f4b6993582e283630727a38e5

@mgp25 We may want to do another hotfix along with your other upcoming commit as this is a larger issue
